docs/ContainerizedSetup.asciidoc

```diff
@@ -9,65 +9,72 @@ This section describes two ways to deploy the containers for the openQA web UI
+
+Our intent was to create universal `webui` and `worker` containers and move
+all data storage and configurations to a third container, called `openqa_data`.
+`openqa_data` is a so called
+https://docs.docker.com/storage/volumes#creating-and-mounting-a-data-volume-container[Data Volume Container]
+and is used for the database and to store results and configuration. During
+development and in production, you could update `webui` and `worker` images
+but as long as `openqa_data` is intact, you do not lose any data.
+
+To make development easier and to reduce the final size of the `openqa_data`
+container, this guide describes how to override `tests` and `factory`
+directories with directories from your host system. This is not necessary but
+recommended. This guide is written with this setup in mind.
+
+It is also possible to use `tests` and `factory` from within the `openqa_data`
+container (so you do not have any dependency on your host system) or to leave
+out the `openqa_data` container altogether (so you have only `webui` and
+`worker` containers and data is loaded and saved completely into your host
+system). If this is what you prefer, checkout the sections
+<<ContainerizedSetup.asciidoc#_keeping_all_data_in_the_data_volume_container,Keeping all data in the Data Volume Container>>
+and
+<<ContainerizedSetup.asciidoc#_keeping_all_data_on_the_host_system,Keeping all data on the host system>>
+respectively.
+
+Otherwise, when you want to have the big files (isos and disk images, tests and
+needles) outside of the Volume container, you should create this file
+structure from within the directory you are going to execute the container:
+
+    mkdir -p data/factory/{iso,hdd} data/tests
+
+It could be necessary to either run all containers in privileged mode or to
+setup SELinux properly. If you are having problems with it, run this command:
+
+    chcon -Rt svirt_sandbox_file_t data
 
 == Get container images
 
-You can either build the images locally or get the images from Docker Hub.
-Using the Docker Hub can be easier for a first try.
+You can either build the images locally or use Fedora images from Docker Hub.
 
-=== Download images from the Docker Hub
+For the `docker-compose` setup it is required to build the images locally.
 
-==== Fedora images (Fedora)
+=== Download Fedora-based images from the Docker Hub
 
     docker pull fedoraqa/openqa_data
     docker pull fedoraqa/openqa_webui
     docker pull fedoraqa/openqa_worker
 
-=== Build images locally (mandatory when using docker-compose)
-
-The Dockerfiles included in this project are for openSUSE:
+=== Build openSUSE-based images locally (mandatory when using docker-compose)
 
-    # Fedora and docker-compose
     docker build -t openqa_data ./openqa_data
     docker build -t openqa_webui ./webui
     docker build -t openqa_worker ./worker
 
-    # docker-compose only
+    # for docker-compose setup
     docker build -f webui/Dockerfile-lb -t openqa_webui_lb ./webui
```

Comment:
I think we should mention that the images can be built using docker-compose build

Reply:
Or see my next comment, docker-compose is going to build the images if don't exist when call docker-compose up